src/expr/src/scalar/mod.rs

```diff
@@ -548,7 +548,9 @@ impl ScalarExpr {
             // Nullary functions must be transformed away before evaluation.
             // Their purpose is as a placeholder for data that is not known at
             // plan time but can be inlined before runtime.
-            ScalarExpr::CallNullary(_) => panic!("eval called on nullary function"),
+            ScalarExpr::CallNullary(_) => Err(EvalError::Internal(
```

Comment:
@frankmcsherry I think you'll like this change. Just makes things less panicky if we get it wrong in the future.

Reply:
yes, this is great!